doctrine/registration_form.rst

```diff
@@ -178,207 +126,83 @@ the :doc:`Security Guide </security>`.
-    use Symfony\Component\Form\Extension\Core\Type\TextType;
-    use Symfony\Component\Form\Extension\Core\Type\RepeatedType;
-    use Symfony\Component\Form\Extension\Core\Type\PasswordType;
+The controller builds the form and, on submit, encodes the plain password and
+saves the user::
 
-    class UserType extends AbstractType
-    {
-        public function buildForm(FormBuilderInterface $builder, array $options)
-        {
-            $builder
-                ->add('email', EmailType::class)
-                ->add('username', TextType::class)
-                ->add('plainPassword', RepeatedType::class, array(
-                    'type' => PasswordType::class,
-                    'first_options'  => array('label' => 'Password'),
-                    'second_options' => array('label' => 'Repeat Password'),
-                ))
-            ;
-        }
-
-        public function configureOptions(OptionsResolver $resolver)
-        {
-            $resolver->setDefaults(array(
-                'data_class' => User::class,
-            ));
-        }
-    }
-
-There are just three fields: ``email``, ``username`` and ``plainPassword``
-(repeated to confirm the entered password).
-
-.. tip::
-
-    To explore more things about the Form component, read the
-    :doc:`/forms` guide.
-
-Handling the Form Submission
-----------------------------
-
-Next, you need a controller to handle the form rendering and submission. If the
-form is submitted, the controller performs the validation and saves the data
-into the database::
-
-    // src/Controller/RegistrationController.php
     namespace App\Controller;
 
-    use App\Form\UserType;
     use App\Entity\User;
+    use App\Form\RegistrationFormType;
+    use App\Security\StubAuthenticator;
     use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
     use Symfony\Component\HttpFoundation\Request;
+    use Symfony\Component\HttpFoundation\Response;
     use Symfony\Component\Routing\Annotation\Route;
     use Symfony\Component\Security\Core\Encoder\UserPasswordEncoderInterface;
+    use Symfony\Component\Security\Guard\GuardAuthenticatorHandler;
 
     class RegistrationController extends AbstractController
     {
         /**
-         * @Route("/register", name="user_registration")
+         * @Route("/register", name="app_register")
          */
-        public function register(Request $request, UserPasswordEncoderInterface $passwordEncoder)
+        public function register(Request $request, UserPasswordEncoderInterface $passwordEncoder): Response
         {
-            // 1) build the form
-            $user = new User();
-            $form = $this->createForm(UserType::class, $user);
-
-            // 2) handle the submit (will only happen on POST)
+            $form = $this->createForm(RegistrationFormType::class);
             $form->handleRequest($request);
+
             if ($form->isSubmitted() && $form->isValid()) {
+                /** @var User */
```

Comment:
About this line of code, I think I agree with @lyrixx's comment here: https://github.com/symfony/maker-bundle/pull/333#issuecomment-449399522

Being explicit means just 1 additional line of code, but the resulting code looks better and easier to understand, right?

Reply:
I also like the suggestion - I've already updated the docs for the new code that's in this PR: https://github.com/symfony/maker-bundle/pull/343